### DISCUSSION TITLE
Shine ore resourcepack lag

### DISCUSSION THREAD

**Discussion by Bitlod** [MoreMcmeta/core#101] [Core Dev Questions] [answered]
I want to upload this mod for shiny ores for now, because i have build like Embedium + oculus, but it lag so much. Could u fix that issue? I don't think that turnin off resourcepacks is the best idea for mod which allow resourcepacks to use shine custom textures:/
upd: tested without ETF and EMF: Entity models and some textures reading so bad: maybe it better to do this mod smf like EMF add-on, but for custom shiny textures. With add-on for shiny textures by some cool guy moremcmeta works properly, but lag so much and couldn't render entity models...

**Comment by soir20** [comment]
This is likely related to this known issue: https://github.com/MoreMcmeta/core/issues/71. Keep in mind that I don't think removing resource packs is actually a good solution. It's just that removing offending packs can stop the performance issue until it can be addressed; the fix does not look to be simple for me to implement.

Can you please upload or add a link to your resource pack so that I can reproduce this? Furthermore, what graphics card are you using? I'm curious if there's an AMD vs. Nvidia difference here.

**Reply by Bitlod** [reply]
Can't find my 3D version because i'm using huge resourcepack and i think the problem only with ore because i've checked mod work without any mob(also without ETF double render) and it still lagging. I found that version, that pretty close to mine.
Thanks a lot for your help here. I really want to use your mod for my better game experience and i hope you could fix that issue.
P.S: if it would lag only because of 3D shine texture, ill try to put all ore textures back together

[glowing-ores.zip](https://github.com/user-attachments/files/17601894/glowing-ores.zip)

**Reply by soir20** [reply]
I've released MoreMcmeta 4.5.0 with a potential fix for the performance issue. Can you please check whether the new version improves performance with your resource pack?

I think I would need the animated 3D overlays to check this; it's the animations themselves and not simply the overlays that cause the lag.

**Reply by Bitlod** [reply]
I've tested ur new version and it now lag less more than in previous ver. but its great upgrade
i playing with built-in Intel Graphics core i5 and it works well on 50 fps(as my max), but with moremcmeta mod it renders too slow and can lag to 1 fps because of rendering textures, that using emissive textures like my shine ores pack
ill add my shine ores pack in the next comment, i need to take it separatly from the big one
Also thanks for your interest in it!
![image](https://github.com/user-attachments/assets/4ed8f9b1-0025-49e0-9b5f-fd7c876ce5ab)

**Reply by Bitlod** [reply]
Thats the 3d texture blocks i use with mod that allows to use Optifine packs with your mod(maybe that mod just using resoursepack files but with .moremcmeta format)
I hope that will help you :p
[shine ores 3d test.zip](https://github.com/user-attachments/files/17730225/shine.ores.3d.test.zip)

**Reply by soir20** [reply]
Just to clarify, you are seeing an improvement in 4.5.0? Roughly how much of an improvement do you see (on average)?

**Reply by Bitlod** [reply]
Yes, in 4.5.0 your mod renders textures faster and it don't lag in uploaded chunks, but still using many computer resources for creating new chunks. Maybe it's 10% of Optifine's shine texture render, which creating chunks with that ore with no lag, no offense.

**Reply by soir20** [reply]
Ah, so you only see the lag when moving into new chunks? When you stand still, the performance is OK?

**Reply by soir20** [reply, marked as answer]
~~I'm curious if you still see the poor chunk loading performance if you remove Embeddium. I did some testing and it seems like there might be an interaction between MoreMcmeta and Embeddium causing the issue. Without Embeddium, the chunks load about as quickly as vanilla (and some very basic profiling indicates there's only like a 6ms difference at most).~~ This chunk-uploading issue seems to be specific to the emissive plugin; I had thought it was an issue with core.

**Reply by soir20** [reply]
The chunk loading issue is also definitely worse in 1.20.1 than 1.21.1

**Reply by Bitlod** [reply]
Oh, i see
Can you fix emissive plugin for better render? Or its unfixible problem in 1.20.1 version?
And yes, while i'm not moving, it works good in 4.5.0

**Reply by soir20** [reply]
I'll see what can be done to improve the emissive plugin performance. If you haven't already, I would test switching your chunk loading mode to "threaded" in the game's video settings. I still see the issue even with my game set to threaded chunk loading, but maybe you will see a slight improvement.

**Reply by soir20** [reply]
Do you notice the chunk upload lag when you don't have any resource packs applied?

**Reply by Bitlod** [reply]
Ye, it doesn't lag without any resourcepack.
Also when i create new world my license minecraft crashes...

**Reply by soir20** [reply]
> Ye, it doesn't lag without any resourcepack.

Do you have emissive overlays applied to any fairly common blocks? (Stone, dirt, sand, etc.) Keep in mind that the emissive plugin essentially has to render emissive blocks twice (once for the original block and again for the overlay), so I notice the worst performance when adding overlays to common blocks.

Are you able to narrow down the issue to the shiny ore resource pack you shared earlier? Or perhaps there's another resource pack causing the lag? (Please share it if so.)

> Also when i create new world my license minecraft crashes...

It's impossible to say why this would happen without a stack trace/error information, especially if you are using other mods besides MoreMcmeta.

**Reply by Bitlod** [reply]
As I get it, emissive plugin need for rendering textures that use postfix _e, as it said in emissive.properties file. In my resourcepack i have shine creatures/blocks only warden, others dont use _e postfix and using .mcmeta for EMF/ETF 3D rendering. Common blocks are not changed, i know about consequenses of render every block in 3D.
Warden is a rare creature, so, i think, it can't produce lag. Also i've tested mod without any mob, so that lags producing only by ores.
I think shine ore do that lag, i've checked the main issue when i found fps problems. 
Also i don't think that the problem in mods, because that mods using only geckolib library for rendering
